### PR TITLE
Feat/coapplicant pincode ux

### PR DESCRIPTION
**Pull request by MasterKenth**
## Explain the changes you’ve made

Replaced symmetric key exchange with pin code generation/input.

## Explain why these changes are made

To improve stability in encryption between two users. The old symmetric system was prone to errors due to too many edge cases and difficult scenarios.

## Explain your solution

* CaseContext/Actions/Reducer etc. has been updated to use CaseEncryptionService, and check for and generate pins if needed.
* CaseOverview has been updated to handle cases with pins and use PinInputModal for inputting pins.
* Cases are now always displayed in CaseOverview even if they can't be decrypted (however button actions are disabled then).
* Code related to polling has been (mostly) purged.
* Removed DEPRECATED encryption files.

## How to test

Concrete example:

1. Checkout this branch
2. Verify that solo applicant still works as usual.
3. Verify that main applicant can start applications without waiting for any "confirmation"
4. Verify that co-applicant can follow the case progress but not act on it (no button)
5. Verify that after main applicant has signed, co applicant can input pin code. Pin code is displayed on (main applicant) case overview.
6. Verify that co-applicant, after inputting pin, can delete data/re-install app and still input pin and continue as normal.
7. Verify that the case is submitted correctly after co-applicant has signed.

## Tested environments

- [] Storybook on a iOS device/simulator.
- [x] Building the Application on a iOS device/simulator.
- [] Building the Application on a Android device/simulator.

## Screenshots

Main applicant:
image

Co-applicant:
image

image

image

image

---

## Checklist for reviewers

Here are a few points that are useful for reviewers to check.

The short list of the (maybe) most important points are:

- Does this PR work as expected?
- Does the PR properly only concern a single feature/bug/thing?
- Is the code clear and concise?
- Will the changes impact existing functionality?
- Are there any linting errors (ESLint) that should be resolved?
- Do the automatic GitHub checks pass?

However it is good if a reviewer can be as thorough as possible,
within reasonable limits of course.

### Functionality

- Does this PR work as expected?
- If this is a new feature, does it add value, or is it a sign of feature-creep?
- Will the changes impact existing functionality?
- Does the changes create inconsistencies somewhere else?
- Does the code work on different platforms if applicable?
- How does the code handle if unexpected values are passed to functions etc.?

### Readability, syntax, formatting, style

- Is the code clear and concise?
- Has styling rules (Prettier) been applied to included files?
- Are functions/variables/etc. aptly named?

### Design

- Is the code properly planned and designed?
- Is code organized/separated in a logical fashion?
- Does the code follow clean code practices to a reasonable degree?
- Is the code self-documenting enough?
- Is the code in sync with existing patterns/technologies?
- Does the code reuse existing functionality where applicable?
- Is the code designed with reusability in mind?
- Does the code have hard-coded values that can be refactored?

### Patterns, idioms, tests, best practices

- Are there any linting errors (ESLint) that should be resolved?
- Is the code properly typed (TypeScript)?
- Does the code utilize language features to their full potential?

### Debugging, testing, configuring

- Is appropriate logging used?
- Have tests been added/modified to accommodate the changes?
- Is the feature sufficiently configurable in case tweaks are required in the future?

### Performance, reliability, scalability

- Is the code well optimized?
- Does the code scale?
- Does the code handle failure scenarios?

### Security

- Is there any obvious potential for exploits and security threats?
- Does code check proper authorization where applicable?

### Etiquettes

- Are commits atomic (does only one thing per commit)?
- Does the PR properly only concern a single feature/bug/thing?
- Are commit messages well-written?
- Do the automatic GitHub checks pass?

### CODE REVIEW

**Comment by sebastianthulin** [comment]
Quality gate Quality gate: **_Passed_** 
 


            
 No issues Everything looks good
 [Details](https://app.embold.io/organization/gh/helsingborg-stad/repositories/fedfce3fd8695f238dea9cb00bd37ce7/pull-request/661)

**Comment by helsingborgsstad** [comment]
Task linked: [CU-21w37tg Visa ärenden i listan även om de inte går att avkryptera](https://app.clickup.com/t/21w37tg)

**Comment by helsingborgsstad** [comment]
Task linked: [CU-22ydyxr Visa modal för medsökande för att mata in pinkod](https://app.clickup.com/t/22ydyxr)

**Comment by helsingborgsstad** [comment]
Task linked: [CU-21w388k Använd rätt krypteringsform för ärenden när de hämtas/uppdateras](https://app.clickup.com/t/21w388k)

File: source/screens/caseScreens/CaseOverview.tsx [hunk 1]
```diff
@@ -504,28 +532,7 @@ function CaseOverview(props): JSX.Element {
...
-      const isWaitingForCoApplicantConfirm =
-        currentForm.encryption.publicKeys &&
-        !caseData.hasSymmetricKey &&
-        !Object.entries(currentForm.encryption.publicKeys).every(
-          (item) => item[1] !== null
-        );
-
-      return isWaitingForCoApplicantConfirm;
-    });
+    const coApplicantItemsToSign: CaseWithExtra[] = [];
```
Comment (karlknowit) [inline source/screens/caseScreens/CaseOverview.tsx:535]: This can be removed

File: source/screens/caseScreens/CaseOverview.tsx [hunk 2]
```diff
@@ -187,29 +200,25 @@ const computeCaseCardComponent = (caseData, navigation, authContext, extra) => {
...
-    : isOngoing ||
-      isNotStarted ||
-      isRandomCheckRequired ||
-      isSigned ||
-      isClosed ||
-      isVivaCompletionRequired ||
-      isCompletionSubmitted;
+  const currentForm: AnsweredForm = caseData?.forms[caseData.currentFormId];
+  const selfNeedsToConfirm = false;
+  const isWaitingForCoApplicantConfirm = false;
```
Comment (karlknowit) [inline source/screens/caseScreens/CaseOverview.tsx:205]: these can be removed

**Review by karlknowit — CHANGES_REQUESTED** [review]

**Review by karlknowit — CHANGES_REQUESTED** [review]

File: source/components/organisms/PinInputModal/styled.ts [hunk 3]
```diff
@@ -12,3 +13,9 @@ export const ButtonContainer = styled.View`
   margin-top: 25px;
   width: 100%;
 `;
+
+export const ErrorText = styled(Text)`
+  text-align: center;
+  margin-top: 10px;
+  color: #e14751;
```
[the remark this answers is not in the document]
Comment (MasterKenth) [reply source/components/organisms/PinInputModal/styled.ts:20]: That exact color (as specified in Figma) is not in the palette, but I'll change to use a close-enough color from the palette for now, so it's a bit cleaner.

**Review by karlknowit — APPROVED** [review]
image

**Pull request merged**